File: 443.string-compression.cpp
#include <bits/stdc++.h>
using namespace std;
/*
 * @lc app=leetcode id=443 lang=cpp
 *
 * [443] String Compression
 */

// @lc code=start
class Solution
{
public:
    int compress(vector<char> &chars)
    {
        int idx = 0, i = 0;

        while (i < chars.size())
        {
            int j = i;

            while (j < chars.size() && chars[j] == chars[i])
            {
                j++;
            }

            chars[idx++] = chars[i];

            int occ = j - i;
            if (occ > 1)
            {
                string str = to_string(occ);
                for (char c : str)
                {
                    chars[idx++] = c;
                }
            }
            i = j;
        }
        return idx;
    }
};
// @lc code=end

File: Java/Daily Problems/March2023/443.string-compression.java
package March2023;
/*
 * @lc app=leetcode id=443 lang=java
 *
 * [443] String Compression
 */

// @lc code=start
class Solution {
    public int compress(char[] chars) {
        int res = 0, count = 0;
        char curr = chars[0];
        for (int i = 0; i < chars.length; i++) {
            if (curr != chars[i]) {
                chars[res++] = curr;
                if (count > 1) {
                    for (char ch : Integer.toString(count).toCharArray()) {
                        chars[res++] = ch;
                    }
                }
                count = 0;
                curr = chars[i];
            }
            count++;
        }
        chars[res++] = curr;
        if (count > 1) {
            for (char ch : Integer.toString(count).toCharArray()) {
                chars[res++] = ch;
            }
        }
        return res;
    }
}